Make defense missions have a deadline (#8905)
endless-sky/endless-sky: data/human/south jobs.txt

	name `Pirates occupying <system>`
	minor
	repeat

> 	description `A small band of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system before <deadline> and land on <planet> when done. Payment is <payment>.`

Author: Doing the second one, thanks!

> 	description `A small band of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <deadline>, to receive the payment of <payment>.`

Reviewer: If the deadline is the day after shouldnt we mention it that way?
Also none of these descriptions are too long

Author: No, sorry, I haven't checked the length. I'll get to that soon.

Can you explain what you're asking with the deadline with a suggested change? I'm not sure what you mean.

Author: It's not too long, but the substitutions don't work. (I changed it on my save to <date> but it made no difference.)
<img width="1680" alt="Screen Shot 2023-07-01 at 7 15 59 AM" src="https://github.com/endless-sky/endless-sky/assets/125756761/b3e93bda-7d48-4ff0-9a18-a305fac2a7cf">

Reviewer: ```suggestion
	description `A small band of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> in the same day, to receive the payment of <payment>.`
```
I meant smth like this
Also if it doesnt work its probably because of the wrong substitution being used?

Author: The `before <deadline>` refers to both landing and when you get there.
And I've used `<deadline>` and `<date>` and no substitutions.

> 	description `A small band of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <deadline>, to receive the payment of <payment>.`

Reviewer 2: I don't know why I said `deadline`.
```suggestion
	description `A small band of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <date>, to receive the payment of <payment>.`
```

Author: It's fine, I can't remember which works and which doesn't.

	name `Pirates occupying <system>`
	minor
	repeat

> 	description `A gang of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <deadline>, to receive the payment of <payment>.`

Reviewer 2: ```suggestion
	description `A gang of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <date>, to receive the payment of <payment>.`
```

	name `Pirates occupying <system>`
	minor
	repeat

> 	description `A large gang of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <deadline>, to receive the payment of <payment>.`

Reviewer 2: ```suggestion
	description `A large gang of pirates have been occupying <system>, and the local government has asked for help. Defeat the pirates in the system and land on <planet> before <date>, to receive the payment of <payment>.`
```
